src/client/Client.cc

```diff
@@ -7742,8 +7829,15 @@ int Client::path_walk(InodeRef dirinode, const filepath& origpath, walk_dentry_r
     dn = get_or_create(diri.get(), dname.c_str());
 
     /* Get extra requested caps on the last component */
-    if (i == (path.depth() - 1))
+    if (i == (path.depth() - 1)) {
       caps |= extra_options.mask;
+      if (diri->is_fscrypt_enabled()) {
```

Comment:
ifdef this out until we figure out what caused the test failure

Reply:
I believe this is working now, resolved.